internal/alloy/logging/logger.go

```diff
@@ -93,15 +98,29 @@ func NewDeferred(w io.Writer) (*Logger, error) {
 			w:         &writer,
 			leveler:   &leveler,
 			formatter: &format,
+			replacer:  replace,
 		},
 	}
+	l.deferredSlog = newDeferredHandler(l)
+
+	return l, nil
+}
+
+// newDeferredTest creates a new logger with the default log level and format. Used for tests.
+// The logger is not updated during initialization.
+func newDeferredTest(w io.Writer) (*Logger, error) {
+	l, err := NewDeferred(w)
+	if err != nil {
+		return nil, err
+	}
+	l.handler.replacer = testReplace
 
 	return l, nil
 }
```

Comment:
nit: if it's only used in tests, maybe this should be defined in a test file instead?